filter_plugins/netdata.py

```diff
@@ -19,3 +20,27 @@ def aws_creds(self, data, type_='access'):
         if type_ == 'secret':
             return config.get('default', 'aws_secret_access_key')
         return config.get('default', 'aws_access_key_id')
+
+    def redis_roles(self, services):
+        """
+        Organise redis services into clusters for the redis collector
+        ---
+        Example:
+        services = [
+            {"ip": "10.10.10.11", "port": 6011, "role": "account"},
+            {"ip": "10.10.10.11", "port": 6411, "role": "ch"},
+            {"ip": "10.10.10.11", "port": 6511, "role": "bdb"}
+        ]
+        self.redis_roles(services)
+        > "10.10.10.11:6011:account,10.10.10.11:6411:ch,10.10.10.11:6511:bdb"
+        """
+        tmp = dict()
+        res = []
+        for service in services:
+            tmp.setdefault(service['role'], [])
+            tmp[service['role']].append(service)
+        for idx, cluster in enumerate(tmp.values()):
+            for service in cluster:
+                res.append("%s:%d:%d" %
+                           (service['ip'], service['port'], service['role']))
```

Comment:
error `service['role']` is a string but you try to export it to an int `%d`. The line should be fixed as follow:
 ```
                res.append("%s:%d:%s" %
                           (service['ip'], service['port'], service['role']))
```

Reply:
fixed by @vdombrovski